### DISCUSSION TITLE
Slope background

### DISCUSSION THREAD

**Discussion by SHilgers** [Julian-Hochhaus/LG4X-V2#49] [Ideas]
For some type of data it would be nice to add a slope background which account for the change in the slope. 
For further information about the slope background see [https://doi.org/10.1016/j.elspec.2013.07.006]

The slope function is very similar to that for the shirley background. 
The slop could be implemented like this:

> 
```
def slope(x, y, k, const):
    n=len(y)
    #print('len slope',n)
    y_right=const
    y_temp=y-y_right 
    bg=[]
    bg2=[]
    for i in range(n):
        bg.append(np.sum(y_temp[i:]))  #sum from i until end of y_temp
    bg=[elem+y_right for elem in bg]
    for j in range(len(bg)):
        bg2.append(np.sum(bg[j:]))
    return [-k*elem for elem in bg2]
```

**Comment by SHilgers** [comment]
I have also tested a slightly different definition of the slope(see below). The difference is, that the slope has no longer the parameter 'const', the only parameter is k_slope. 
When using the slope together with a shirley, i have seen nearly no difference between both variants of slope.  
When using it with tougaart this variant was a little bit better  for my test. 


```
def slope(y, k):
    n=len(y)
    y_right=np.min(y)           
    y_temp=y-y_right 
    temp=[]
    bg=[]
    for i in range(n):
        temp.append(np.sum(y_temp[i:]))     
    for j in range(n):
        bg.append(np.sum(temp[j:]))        
    return np.asarray([-k*elem for elem in bg])
```

**Reply by Julian-Hochhaus** [reply]
Was implemented in the `lmfitxps` package which is used in `LG4x-V2`.
Thanks for the idea!
